### PR TITLE
Feature/add delete diagrams

### LINKED ISSUE #276: Multi-diagrams page: delete diagram
## Description

On click on a diagram, put the border of diagram in accent, display an action bar above the mouse, the action can't move it's fixed.

 => If we click on diagram that is already selected, remove accent color and the action bar. We can select only one diagram.

In this action bar, add a button to delete diagram.

When user click on this button, a popup confirmation appear to validate the diagram deletion.

Once diagram is deleted, make a toast appear to confirm the deletion.

If #275 is done, you can only select/unselect diagram when focus is not activated. And if diagram is activated, unselect diagram.

:warning: keep the actual design and  position of toaster and dialog.

## Mockup

image
image
image

### PR DESCRIPTION
**Pull request by k-y0u**
resolves #276

### CODE REVIEW

File: src/pages/DiagramsPage.vue [hunk 1]
```diff
@@ -3,6 +3,7 @@
     id="diagrams-page"
     v-touch-pan.mouse="handlePan"
     data-cy="diagrams-page"
+    @click="selecteDiagram()"
```
Comment (justinehell) [inline src/pages/DiagramsPage.vue:6]: franglish x)

File: src/pages/DiagramsPage.vue [hunk 2]
```diff
@@ -11,13 +12,33 @@
...
           :data-cy="`diagram_${diagram.path}`"
         />
+        <q-popup-proxy>
+          <q-btn-group>
+            <q-btn
+              stack
+              size="sm"
+              icon="fa-solid fa-trash"
+              :label="$t('page.diagrams.actions.delete')"
+              data-cy="delete-btn"
```
Comment (justinehell) [inline src/pages/DiagramsPage.vue:37]: btn ? :O button please ;)

File: src/pages/DiagramsPage.vue (hunk 1 above)
Comment (justinehell) [reply src/pages/DiagramsPage.vue:6]: don't forget to change inside the test

File: changelog.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 * Multi diagrams page:
   * Focus movement on all diagrams.
+  * Delete diagrams.
```
Comment (justinehell) [inline changelog.md:14]: Add delete diagrams actions. ?

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/pages/DiagramsPage.vue (hunk 2 above)
Comment (k-y0u) [reply src/pages/DiagramsPage.vue:37]: yes madame...

File: tests/unit/pages/DiagramsPage.spec.js [hunk 4]
```diff
@@ -51,6 +72,46 @@ describe('Test page component: DiagramsPage', () => {
...
+  describe('Test function: selecteDiagram', () => {
+    it('should set "selectedDiagram" to null without parameter', () => {
+      wrapper.vm.selectedDiagram = 'test';
+
+      wrapper.vm.selecteDiagram();
+
+      expect(wrapper.vm.selectedDiagram).toBeNull();
+    });
+
+    it('should set "selectedDiagram" to null with parameter is selected diagram', () => {
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply tests/unit/pages/DiagramsPage.spec.js:84]: done

File: tests/unit/pages/DiagramsPage.spec.js [hunk 5]
```diff
@@ -51,6 +72,46 @@ describe('Test page component: DiagramsPage', () => {
...
+      wrapper.vm.selectedDiagram = 'test';
+
+      wrapper.vm.selecteDiagram('test');
+
+      expect(wrapper.vm.selectedDiagram).toBeNull();
+    });
+
+    it('should set "selectedDiagram" to diagram in parameter', () => {
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply tests/unit/pages/DiagramsPage.spec.js:92]: done

File: src/components/table/DiagramTable.vue [hunk 6]
```diff
@@ -80,6 +80,7 @@ import DiagramTableActionMenu from 'src/components/menu/DiagramTableActionMenu.v
 
 const { t } = useI18n();
 const emit = defineEmits(['click:diagram']);
+
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply src/components/table/DiagramTable.vue:83]: I put typos cause it was one line to add & one to remove. No spaces involved.

File: cypress/e2e/DiagramsPage/DeleteDiagrams.feature [hunk 7]
```diff
@@ -0,0 +1,43 @@
+Feature: Test diagrams page: delete diagram
+  Scenario: Should delete diagram
+    Given I clear cache
+    And   I set viewport size to '1920' px for width and '1080' px for height
+    And   I visit the '/'
+    And   I set context field 'projectName' with 'projectTest'
+    And   I set context field 'modelPath' with 'infra/main.tf'
```
Comment (Zorin95670) [inline cypress/e2e/DiagramsPage/DeleteDiagrams.feature:7]: modelPath -> modelFile

File: cypress/e2e/DiagramsPage/ZoomDiagrams.feature [hunk 8]
```diff
@@ -31,7 +31,8 @@ Feature: Test diagrams page: zoom on all diagrams
 
     # Back to the diagrams page
     When I visit the 'localhost:8080/#/projects/{{ projectName }}/diagrams'
-    And I wait 2 seconds
```
Comment (Zorin95670) [inline cypress/e2e/DiagramsPage/ZoomDiagrams.feature:34]: put this in another commit like `fix e2e zoom test`

File: cypress/e2e/DiagramsPage/DeleteDiagrams.feature [hunk 9]
```diff
@@ -0,0 +1,43 @@
+Feature: Test diagrams page: delete diagram
```
Comment (Zorin95670) [inline cypress/e2e/DiagramsPage/DeleteDiagrams.feature:1]: put this in commit like "Update e2e test"

File: src/i18n/en-US/index.js [hunk 10]
```diff
@@ -269,6 +269,7 @@ export default {
       actions: {
```
Comment (Zorin95670) [inline src/i18n/en-US/index.js:269]: Put this in commit "Update translation"

File: src/pages/DiagramsPage.vue [hunk 11]
```diff
@@ -68,8 +93,33 @@ const translate = reactive({
   x: 0,
   y: 0,
 });
+const selectedDiagram = ref(null);
 
 let pluginInitSubscription;
+let updateModelSubscription;
+
+/**
+ * Select or unselect diagram.
```
Comment (Zorin95670) [inline src/pages/DiagramsPage.vue:102]: `Toggle diagram selection`

File: changelog.md (hunk 3 above)
Comment (Zorin95670) [inline changelog.md:14]: Add diagram selection too

File: src/pages/DiagramsPage.vue [hunk 12]
```diff
@@ -3,6 +3,7 @@
     id="diagrams-page"
```
Comment (Zorin95670) [inline src/pages/DiagramsPage.vue:3]: keep diagramsPage in commit `Add delete diagram`. And rename to `Add delete diagram action`

File: cypress/e2e/DiagramsPage/DeleteDiagrams.feature (hunk 7 above)
Comment (k-y0u) [reply cypress/e2e/DiagramsPage/DeleteDiagrams.feature]: done

File: cypress/e2e/DiagramsPage/ZoomDiagrams.feature (hunk 8 above)
Comment (k-y0u) [reply cypress/e2e/DiagramsPage/ZoomDiagrams.feature:34]: done

File: cypress/e2e/DiagramsPage/DeleteDiagrams.feature (hunk 9 above)
Comment (k-y0u) [reply cypress/e2e/DiagramsPage/DeleteDiagrams.feature:1]: done

File: src/i18n/en-US/index.js (hunk 10 above)
Comment (k-y0u) [reply src/i18n/en-US/index.js:269]: done

File: src/pages/DiagramsPage.vue (hunk 11 above)
Comment (k-y0u) [reply src/pages/DiagramsPage.vue]: done

File: changelog.md (hunk 3 above)
Comment (k-y0u) [reply changelog.md:14]: done

File: src/pages/DiagramsPage.vue (hunk 12 above)
Comment (k-y0u) [reply src/pages/DiagramsPage.vue:3]: done

**Review by Zorin95670 — DISMISSED** [review]

**Review by justinehell — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]
